src/qt/bitcoingui.cpp

```diff
@@ -585,6 +585,9 @@ void BitcoinGUI::setWalletActionsEnabled(bool enabled)
 
 void BitcoinGUI::createTrayIcon(const NetworkStyle *networkStyle)
 {
+    if (!QSystemTrayIcon::isSystemTrayAvailable()) {
```

Comment:
I would prevent calls to `createTrayIcon` and maybe `assert(QSystemTrayIcon::isSystemTrayAvailable())` here?